Gateway/Validator/CheckoutResponseValidator.php

```diff
@@ -105,32 +106,7 @@ public function validate(array $validationSubject)
                     $payment->setAdditionalInformation('pspReference', $response['pspReference']);
                     break;
                 case "PresentToShopper":
-                    $payment->setAdditionalInformation('pspReference', $response['pspReference']);
-                    // set additionalData
-                    if (isset($response['outputDetails']) && is_array($response['outputDetails'])) {
-
-                        $outputDetails = $response['outputDetails'];
-                        if (isset($outputDetails['boletobancario.dueDate'])) {
-                            $payment->setAdditionalInformation(
-                                'dueDate',
-                                $outputDetails['boletobancario.dueDate']
-                            );
-                        }
-
-                        if (isset($outputDetails['boletobancario.expirationDate'])) {
-                            $payment->setAdditionalInformation(
-                                'expirationDate',
-                                $outputDetails['boletobancario.expirationDate']
-                            );
-                        }
-
-                        if (isset($outputDetails['boletobancario.url'])) {
-                            $payment->setAdditionalInformation(
-                                'url',
-                                $outputDetails['boletobancario.url']
-                            );
-                        }
-                    }
+                    $payment->setAdditionalInformation('action', $response['action']);
```

Comment:
Check if $response['action'] is not empty before assigning in the additionalInformation
Also a question for the pspReference, why was it removed from the additionalInformation? Don't we use it anymore?

```suggestion
                    $payment->setAdditionalInformation('pspReference', $response['pspReference']);
                    if (!empty($response['action']) {
                        $payment->setAdditionalInformation('action', $response['action']);
                    }
```

Reply:
Added !empty check on action and pspReference and assigned both variables